### ISSUE TITLE
[REQUEST]: 

### ISSUE THREAD

**Issue by MikeShah** [compiler-explorer/compiler-explorer#5446]
### Is your feature request related to a problem? Please describe

I would like the ability to hide library functions in assembly on https://d.godbolt.org/

### Describe the solution you'd like

Ability to hide (or even just fold) all of the library functions by default.

### Describe alternatives you've considered

Right now I just 'fold all' and then ctrl+f for functions or data structures names. This is a bit challenging as the names are managled, but it would be nice to shrink the search space.

### Additional context

Not applicable

**Comment by jeremy-rifkin** [comment]
Hi @MikeShah, we support library code filtering for a number of languages currently. I think it all relies on debug info, and it looks like D is emitting the necessary debug info for library functions so this should be feasible.

**Comment by MikeShah** [comment]
That would be great! Adding the compiler page here https://dlang.org/dmd-osx.html (Should just be a matter of -g to emit bug info)

**Comment by MikeShah** [comment]
https://dlang.org/dmd-linux.html (Linux page)

**Comment by MikeShah** [comment]
Sure: Here's an example snippet. https://d.godbolt.org/z/d46365fe3

I believe any function that starts with 'std', 'core', or 'etc' may be interesting to filter out:
e.g.

`std.stdio.File.write`

Here's a link to the library if a full list is needed: https://dlang.org/phobos/index.html

In D functions have attributes (e.g. @safe) that perhaps may be preventing them from being hidden.

The above function calls full label is:
@safe void std.stdio.File.write!(int, char).write(int, char):

It appears right now that 'Library Functions' is always checked off (image below), thus lacking the ability to hide :)

<img width="345" alt="image" src="https://github.com/compiler-explorer/compiler-explorer/assets/1079349/9fbd8140-ccd8-46bc-8e88-0ff17023aa46">

**Comment by jeremy-rifkin** [comment]
The library functions filter is greyed out until `supportsLibraryCodeFilter` is specified for the compiler. It'll probably be most robust to rely on debug information rather than function - otherwise name mangling and other third party libraries would be an issue.

**Comment by MikeShah** [comment]
Agreed, probably best to rely on debug information -- let me know if I can provide anything more useful to help.
